src/de/espend/idea/laravel/controller/namespace/ControllerNamespaceCutter.java

```diff
@@ -0,0 +1,17 @@
+package de.espend.idea.laravel.controller.namespace;
+
+import com.intellij.openapi.project.Project;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Cuts the controllers namespace for completion. Usually using prefix and default namespace.
+ */
+public interface ControllerNamespaceCutter {
+    void init(@NotNull Project project, @Nullable String prefix);
```

Comment:
moving method parameters into `cut` or `process` makes this call redundant, see also comment below